File: frontend/src/app/services/comment.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

const BASIC_URL = "http://localhost:8885/api/comments";


@Injectable({
  providedIn: 'root'
})
export class CommentService {

  constructor(private http: HttpClient) { }

  createComment(postId: number, postedBy: string, content: string): Observable<any> {
    const params= {
      postId: postId,
      postedBy: postedBy
    };

    return this.http.post(`${BASIC_URL}/create-comment`, content, {params});

  }

  getCommentsByPost(postId: number): Observable<any> {
    return this.http.get(`${BASIC_URL}/${postId}`);
  }
}

File: BloggingServer/src/main/java/com/project/BloggingServer/Service/CommentService.java
package com.project.BloggingServer.Service;

import com.project.BloggingServer.entity.Comment;

import java.util.List;

public interface CommentService {
    Comment createComment(Long postId, String postedBy, String content);
    List<Comment> getCommentsByPostId(Long postId);
}
